### ISSUE TITLE
reconsider rewriting of global property bindings in residual functions

### ISSUE THREAD

**Issue by NTillmann** [facebook/prepack#918]
Currently, we prepack
```js
(function() {
  x = 42;
  inspect = function() { return x; }
})();
```
to
```js
(function () {
  var _$0 = this;

  function _0() {
    return _$0.x;
  }

  x = 42;
  inspect = _0;
}).call(this);
```

For the purposes of this issue, let's focus on the fact that the reference to `x` in the `inspect` function got rewritten to `_$0.x`. (This happens in the ResidualHeapSerializer.js in `_serializeGlobalBinding` for all references to global names which are not `undefined` and for which there's no (let) binding.)
- Consider not rewriting to `_$0.x`, but just leaving `x` as it is. There should not be any risk of a name collision with anything defined in the enclosing anonymous function scope that Prepack generates, as Prepack never reuses any existing identifier in the program.
- To make the call, evaluate performance and code size consequences: Check with VM folks and by measuring if the `_$0.x` variant is expected to be faster or closer, and if it's increasing (bytecode) size or possibly decreasing it.

**Comment by NTillmann** [comment]
Related: `serializeGlobalBinding` currently turn all references to `undefined` into `void 0`. Does that have any effect on performance and/or bytecode size?

**Comment by NTillmann** [comment]
The main request was implemented by #1387. I also verified that `void 0` has no negative implications.
